### ISSUE TITLE
Add global option to reverse order of legend entries

### ISSUE THREAD

**Issue by evanmascitti** [tidyverse/ggplot2#5761]
---

I often make plots with ordered factors on the y-axis. However the entries appear in the legend in ascending order from to bottom, while the y-axis shows them in descending order from top to bottom. Obviously this is due to factors having an integer attribute which on the plot would increase moving upward. However this is visually undesirable as the colors in the legend are in opposite order from those on the plot:

``` r
library(purrr)
library(ggplot2)
library(tibble)
theme_set(theme_bw())

reverse_legend <- TRUE

df <-tibble(
  x = c(sample(1:10, 10, replace = T), sample(10:20, 10, replace = T)),
  y = factor(paste('Level', rep(LETTERS[1:2], each = 10)))
  )

df %>% 
  ggplot(aes(x, y, color = y))+
  geom_point(size = 3)+
  guides(color = guide_legend('Factor level', reverse = reverse_legend))+
  labs(title = "Shouldn't levels A and B should appear in the same order on the y-axis and in the legend?")
```

![](https://i.imgur.com/vYbULMG.png)

<sup>Created on 2024-03-07 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

I realize it is not realistic to change the default behavior at this mature stage of ggplot2. 
However it is annoying to manually reverse the legend order with `guides(color = guide_legend(reverse = TRUE)` every time I make a plot. 

**What about creating a global option for reversing the legend?** This was previously implemented in the `rel_path` argument to `knitr::include_graphics()`. The default argument for guide_legend(reverse = <argument>) could be changed to `guide_legend(reverse = getOption('ggplot.guides.reverse_legend'), FALSE)` and a call to `match.arg()` inside the source code for `guide_legend()`. If no option is set, the default behavior of `reverse = FALSE` would be preserved. 

This would allow the user to set `options(ggplot.guides.reverse_legend = TRUE)` at the top of a script and all plots generated beneath would have the legend reversed. 

Thank you for your consideration.

**Comment by teunbrand** [comment]
I'm not a fan of setting things through options (and I think that things that currently are set through the options in ggplot2 are unfortunate), but I'm not opposed to exposing this legend setting through the theme. I'd affect colourbars too though.

**Comment by clauswilke** [comment]
@evanmascitti I think for your specific use case the correct course of action is to remove the color legend. It serves no purpose. The y axis serves as implicit legend.

In my data visualization class I always teach my students to remove these redundant legends, and having now looked at hundreds such examples I continue to think this is the right choice.

**Comment by teunbrand** [comment]
Agreed with Claus, see also [this section](https://clauswilke.com/dataviz/redundant-coding.html#designing-figures-without-legends) of his book :)

**Comment by evanmascitti** [comment]
@clauswilke I grasp the redundant mapping concept, but on this particular case I would 'agree to disagree.' I like the color as an extra visual aid. However I have read your book and greatly respect your work on data viz, so I will perhaps re-think.

In any case it is not that big a deal to specify the reverse legend for each plot. Thanks to all 3 for the commentary, I will close this issue. Cheers, 
ECM
